Mathlib/LinearAlgebra/DirectSum/Finsupp.lean

```diff
@@ -68,24 +68,86 @@ theorem finsuppTensorFinsupp_symm_single (i : ι × κ) (m : M) (n : N) :
+
+/-- A variant of `finsuppTensorFinsupp` where the second module is the ground ring. -/
+def finsuppTensorFinsuppRid : (ι →₀ M) ⊗[R] (κ →₀ R) ≃ₗ[R] ι × κ →₀ M :=
+  finsuppTensorFinsupp R M R ι κ ≪≫ₗ Finsupp.lcongr (Equiv.refl _) (TensorProduct.rid R M)
+
+@[simp]
+theorem finsuppTensorFinsuppRid_apply_apply (f : ι →₀ M) (g : κ →₀ R) (a : ι) (b : κ) :
+    finsuppTensorFinsuppRid R M ι κ (f ⊗ₜ[R] g) (a, b) = g b • f a := by
+  simp [finsuppTensorFinsuppRid]
+
+@[simp]
+theorem finsuppTensorFinsuppRid_single_tmul_single (a : ι) (b : κ) (m : M) (r : R) :
+    finsuppTensorFinsuppRid R M ι κ (Finsupp.single a m ⊗ₜ[R] Finsupp.single b r) =
+      Finsupp.single (a, b) (r • m) := by
+  simp [finsuppTensorFinsuppRid]
+
+@[simp]
+theorem finsuppTensorFinsuppRid_symm_single (i : ι × κ) (m : M) (r : R) :
+    (finsuppTensorFinsuppRid R M ι κ).symm (Finsupp.single i (r • m)) =
+      Finsupp.single i.1 m ⊗ₜ Finsupp.single i.2 r :=
+  Prod.casesOn i fun _ _ =>
+    (LinearEquiv.symm_apply_eq _).2 (finsuppTensorFinsuppRid_single_tmul_single ..).symm
+
 /-- A variant of `finsuppTensorFinsupp` where both modules are the ground ring. -/
 def finsuppTensorFinsupp' : (ι →₀ R) ⊗[R] (κ →₀ R) ≃ₗ[R] ι × κ →₀ R :=
-  finsuppTensorFinsupp R R R ι κ ≪≫ₗ Finsupp.lcongr (Equiv.refl _) (TensorProduct.lid R R)
+  finsuppTensorFinsuppLid R R ι κ
 #align finsupp_tensor_finsupp' finsuppTensorFinsupp'
 
 @[simp]
 theorem finsuppTensorFinsupp'_apply_apply (f : ι →₀ R) (g : κ →₀ R) (a : ι) (b : κ) :
-    finsuppTensorFinsupp' R ι κ (f ⊗ₜ[R] g) (a, b) = f a * g b := by
-  simp [finsuppTensorFinsupp']
+    finsuppTensorFinsupp' R ι κ (f ⊗ₜ[R] g) (a, b) = f a * g b :=
+  finsuppTensorFinsuppLid_apply_apply R R ι κ f g a b
 #align finsupp_tensor_finsupp'_apply_apply finsuppTensorFinsupp'_apply_apply
 
 @[simp]
 theorem finsuppTensorFinsupp'_single_tmul_single (a : ι) (b : κ) (r₁ r₂ : R) :
     finsuppTensorFinsupp' R ι κ (Finsupp.single a r₁ ⊗ₜ[R] Finsupp.single b r₂) =
-      Finsupp.single (a, b) (r₁ * r₂) := by
-  ext ⟨a', b'⟩
-  classical
-  aesop (add norm [Finsupp.single_apply])
+      Finsupp.single (a, b) (r₁ * r₂) :=
+  finsuppTensorFinsuppLid_single_tmul_single R R ι κ a b r₁ r₂
 #align finsupp_tensor_finsupp'_single_tmul_single finsuppTensorFinsupp'_single_tmul_single
 
-end
+theorem finsuppTensorFinsupp'_symm_single (i : ι × κ) (r₁ r₂ : R) :
+    (finsuppTensorFinsupp' R ι κ).symm (Finsupp.single i (r₁ * r₂)) =
+      Finsupp.single i.1 r₁ ⊗ₜ Finsupp.single i.2 r₂ :=
+  finsuppTensorFinsuppLid_symm_single R R ι κ i r₁ r₂
+
+-- -- cannot add this because we need to import Mathlib.Data.Finsupp.Pointwise
+-- @[simp]
+-- theorem finsuppTensorFinsupp'_symm_single_mul_single (i : ι × κ) (r₁ r₂ : R) :
+--     (finsuppTensorFinsupp' R ι κ).symm (Finsupp.single i r₁ * Finsupp.single i r₂) =
+--       Finsupp.single i.1 r₁ ⊗ₜ Finsupp.single i.2 r₂ := by
+--   rw [← Finsupp.single_mul, finsuppTensorFinsupp'_symm_single]
```

Comment:
Does this import create a loop? If not I think it's appropriate to add it.

Reply:
In fact I doubt the usefulness of this result, in particular as a simp lemma: suppose it was `Finsupp.single i r * Finsupp.single i (s * t)`, then probably `simp` will expand the second `Finsupp.single` and skip this result.

Perhaps we should remove `finsuppTensorFinsupp'_symm_single_mul_single` in favor of `finsuppTensorFinsupp'_symm_single` and asks use to manually `rw` to desired form.

Reply:
I removed this, but added two more results which I need to use later. What's your opinion?